cozy/structures/heaps.py

```diff
@@ -75,6 +78,11 @@ class Heaps(object):
     def owned_types(self):
         return (TMinHeap, TMaxHeap, EMakeMinHeap, EMakeMaxHeap, EHeapElems, EHeapPeek, EHeapPeek2)
 
+    ## TODO: I find it confusing to name one of the formal paramters the
+    ## same as the method itself.  I am not sure which one the occurrence
+    ## in the body refers to, but I think it's the formal parameter.  This
+    ## is confusing.  Is this a standard Python idiom?  I see it in
+    ## `typecheck` below as well.
```

Comment:
`default_value` may need to construct default values for other types---for heaps, for instance, we need a default value for the body of the key projection function. However, we cannot import `construct_value` from evaluation.py (which should be renamed to `default_value`), since that function imports and uses this one! Instead, evaluation.py passes this function a function to call when we need a recursive call. That parameter gets the same name as this function which seemed reasonable; this function constructs a default value for a heap specifically, and the callback constructs a default value for any type.